src/m365/teams/commands/chat/chat-member-add.spec.ts

```diff
@@ -0,0 +1,193 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import { telemetry } from '../../../../telemetry';
+import auth from '../../../../Auth';
+import { Cli } from '../../../../cli/Cli';
+import { CommandInfo } from '../../../../cli/CommandInfo';
+import { Logger } from '../../../../cli/Logger';
+import Command, { CommandError } from '../../../../Command';
+import request from '../../../../request';
+import { pid } from '../../../../utils/pid';
+import { session } from '../../../../utils/session';
+import { sinonUtil } from '../../../../utils/sinonUtil';
+import commands from '../../commands';
+const command: Command = require('./chat-member-add');
+
+describe(commands.CHAT_MEMBER_ADD, () => {
+  const chatId = '19:09fd7575940146d383a4a83fc9598546@thread.v2';
+  const userPrincipalName = 'john@contoso.com';
+  const userId = 'a857e888-b602-4790-86d9-3dca2109449e';
+
+  let log: string[];
+  let logger: Logger;
+  let commandInfo: CommandInfo;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').resolves();
+    sinon.stub(telemetry, 'trackEvent').returns();
+    sinon.stub(pid, 'getProcessName').returns('');
+    sinon.stub(session, 'getId').returns('');
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.post
+    ]);
+  });
+
+  after(() => {
+    sinon.restore();
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.CHAT_MEMBER_ADD);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('adds the member by specifying the userId', async () => {
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/chats/${chatId}/members`) {
+        if (opts.data.roles[0] === 'guest' &&
+          opts.data['user@odata.bind'] === `https://graph.microsoft.com/v1.0/users/${userId}` &&
+          opts.data.visibleHistoryStartDateTime === undefined) {
```

Comment:
While this works, it's not so easy to read. Therefore I suggest that we use an assert to check the request body.
Example:
https://github.com/pnp/cli-microsoft365/blob/05cab2f546df93d908a264a25cd46f0f4243830d/src/m365/aad/commands/user/user-guest-add.spec.ts#L111-L149

This also applies to the other tests